core/build/private.odin

```diff
@@ -0,0 +1,275 @@
+package build
+
+import "core:unicode/utf8"
+import "core:runtime"
+import "core:os"
+import "core:fmt"
+import "core:strings"
+import "core:encoding/json"
+import "core:path/filepath"
+
+// continuation byte?
+_is_cont :: proc(b: byte) -> bool {
+	return b & 0xc0 == 0x80
+}
+
+_utf8_peek :: proc(bytes: string) -> (c: rune, size: int, ok: bool) {
+	c, size = utf8.decode_rune_in_string(bytes)
+
+    ok = c != utf8.RUNE_ERROR
```

Comment:
You seem to be using tabs throughout your project, this line contains two spaces, which looks misaligned to me. Don't thank me :)

Actually looking closely some parts of your code use tabs while other use spaces... you gotta figure it out

Reply:
Hint: Try selecting line 17 and line 19 in github, the selection increments should help you tell spaces and tabs apart, in case you set tab width to 4

Reply:
Could u have a quick look to see if the tab problem was fixed?

Reply:
Looks correct. By the way, which text editor are you using? If it's supported maybe there are options to display invisible characters. I know sublime displays invisible characters when you select text, that sort of functionality may be useful if you're working with projects that have different tab/space rules and you're not sure if your editor messes it up.

Reply:
im using vscode. I haven't been paying much attention to tabs/spaces but it seemed that i was able to just convert the entire file to tabs